shared/models/local_artifact_provider.rb

```diff
@@ -0,0 +1,69 @@
+require_relative "artifact_provider"
+require_relative "artifact"
+require_relative "build"
+require_relative "project"
+
+require "tmpdir"
+require "pathname"
+
+module FastlaneCI
+  # ArtifactProvider backed by a local filesystem.
+  class LocalArtifactProvider < ArtifactProvider
+    class << self
+      # Provide a simple default root_path for users that don't want much configuration.
+      def default_root_path
+        Pathname.new(Dir.tmpdir).join("fastlane-artifacts")
```

Comment:
Instead, it might make sense to make use of `~/.fastlane/artifacts` (we already create `~/.fastlane`) for the user